### PR TITLE
Use new cert flow that doesn't require sudo or public storage containers

### PR DESCRIPTION
**Pull request by ckittel**
* Also fixes the `global` issue for `scheduledQueryRules` (temporary until further research is provided)
* Reduces another `sudo` usage
* Synced deployment scripts to be closer with the new deployment scripts upstream.
* Pointed updated ARM template found upstream from the result of  mspnp/letsencrypt-pip-cert-generation#2

### CODE REVIEW

File: certs/authenticator.sh [hunk 1]
```diff
@@ -1,16 +1,11 @@
...
- --connection-string $STORAGE_CONNECTION_STRING \
- --account-name $STORAGE_ACCOUNT_NAME \
- --container-name verificationdata \
- --name ${CERTBOT_TOKEN}.txt \
- --file ./${CERTBOT_TOKEN}.txt \
- --auth-mode key
+echo "Uploading validation token to $STORAGE_ACCOUNT_NAME"
 
-rm ${CERTBOT_TOKEN}.txt
+az storage blob upload --account-name $STORAGE_ACCOUNT_NAME -c \$web -n "${CERTBOT_TOKEN}" -f "./${CERTBOT_TOKEN}.txt" --auth-mode key --only-show-errors
```
Comment (ferantivero) [inline certs/authenticator.sh:9]: I though .txt extension was no longer required. Please let's just discard this message if not accurate.

File: certs/letsencrypt-pip-cert-generation.sh [hunk 2]
```diff
@@ -1,36 +1,39 @@
 #!/bin/bash -x
 
-SUBDOMAIN=$1
-FQDN=$2
```
Comment (ferantivero) [inline certs/letsencrypt-pip-cert-generation.sh:4]: really nice removal 💪 - resolving this right away.

File: certs/letsencrypt-pip-cert-generation.sh [hunk 3]
```diff
@@ -1,36 +1,39 @@
...
+echo "Existing IP Resource ID: $IP_RESOURCE_ID"
 
-STORAGE_ACCOUNT_NAME=$(az deployment group show -g $RGNAME -n ca-cert-generation --query properties.outputs.storageAccountName.value -o tsv)
-STORAGE_CONNECTION_STRING=$(az storage account show-connection-string -g $RGNAME -n $STORAGE_ACCOUNT_NAME --query "connectionString")
+echo "Creating temporary Resource Group $RGNAME"
+az group create -n $RGNAME -l $LOCATION
 
-echo Storage Account: $STORAGE_ACCOUNT_NAME
-az storage container create --account-name $STORAGE_ACCOUNT_NAME --name verificationdata --auth-mode login --public-access container
+echo "Deploying Azure resources used in validation; this may take 20 minutes."
```
Comment (ferantivero) [inline certs/letsencrypt-pip-cert-generation.sh:19]: why is that required?  `a4f89d43004d4250af02c2bc2194d62467690422`

File: certs/letsencrypt-pip-cert-generation.sh [hunk 4]
```diff
@@ -1,36 +1,39 @@
...
- --name test.txt \
- --file ./test.txt \
- --auth-mode key
+STORAGE_ACCOUNT_NAME=$(az deployment group show -g $RGNAME -n $SUBDOMAIN --query properties.outputs.storageAccountName.value -o tsv)
 
+echo "Enabling web hosting on $STORAGE_ACCOUNT_NAME"
+az storage blob service-properties update --account-name $STORAGE_ACCOUNT_NAME --static-website true --auth-mode login
+
+echo "Uploading placeholder to storage"
+echo pong>ping.txt
```
Comment (ferantivero) [inline certs/letsencrypt-pip-cert-generation.sh:28]: same for `ping.txt` I thought we removed the extension. But please discard the comment again if not accurate. I'm getting confused with the other review apparently.

**Review by ferantivero — APPROVED** [review]
nicely done @ckittel - LGTM 🚢 

just a few comments in case they need to be addressed.

File: certs/authenticator.sh (hunk 1 above)
Comment (ckittel) [reply certs/authenticator.sh:9]: I use the .txt extension locally on the file so that blob storage upload automatically sets the content type.  It's not used in the URL nor is it there in Azure Blob storage, just in the local file.

File: certs/letsencrypt-pip-cert-generation.sh (hunk 4 above)
Comment (ckittel) [reply certs/letsencrypt-pip-cert-generation.sh:28]: Same as answer above.

File: certs/letsencrypt-pip-cert-generation.sh (hunk 3 above)
Comment (ckittel) [reply certs/letsencrypt-pip-cert-generation.sh:19]: Can you say more about `a4f89d43004d4250af02c2bc2194d62467690422` ?  Not sure what I should be looking at.  Was that supposed to be a link?

Comment (ferantivero) [reply certs/letsencrypt-pip-cert-generation.sh:19]: yeah it was the permanent and got mixed up on my brain with  `az blob`. Resolving it now.

Comment (ckittel) [reply certs/letsencrypt-pip-cert-generation.sh:19]: Ah, upstream commit to pin the file version so that we can update upstream without breaking downstream.
